lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -25,7 +25,7 @@ def initialize(expected_error_or_message, expected_message, &block)
           when nil, UndefinedValue
             @expected_error = Exception
             @expected_message = expected_message
-          when String
+          when String, Regexp
```

Comment:
Is there any change in behaviour?
Is looks like that previously a regexp would have been used as an exception?

This change indeed seems the right thing to do. I’m wondering, why it worked the same way before?

Is it because we still were tolerant to whatever it is an exception or something matcheable with a string message?

```ruby
if values_match?(@expected_error, @actual_error) ||
               values_match?(@expected_error, actual_error_message)
```

Reply:
So.. there is a change in "behavior", but it pretty much just affects what failure messages come out. These two ivars are really only used for that purpose, and aren't exposed externally, thankfully; the only impact I can find is that a slightly wrong failure message gets generated (it acts as though you supplied an exception if you supplied only a regex). Could have had larger impact, but the main place we look at the `@expected_error`, we only check `@expected_error != Exception`, which is not true for Regexps either (though I suppose if you supplied `to raise_error(Exception, /foo/)`, you might also be able to get some wrong behavior?).

And the reason the specs didn't mind was just that the case checking this had a test that only checked the part of the failure message the outcomes had in common ("risks false positives"), which was still being produced.